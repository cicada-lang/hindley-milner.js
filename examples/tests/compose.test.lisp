(import id compose "compose.lisp")

id
(id id)
(compose id)
(compose id id)
((compose id id) id)
