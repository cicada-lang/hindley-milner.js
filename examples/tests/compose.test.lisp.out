(claim id (nu (t₁) (-> t₁ t₁)))
(claim compose (nu (t₇ t₈ t₄) (-> (-> t₇ t₈) (-> t₄ t₇) t₄ t₈)))
(the (nu (t₁₀) (-> t₁₀ t₁₀)) (lambda (x) x))
(the (nu (t₂₀) (-> t₂₀ t₂₀)) (lambda (x) x))
(the (nu (t₂₈ t₃₃) (-> (-> t₂₈ t₃₃) t₂₈ t₃₃)) (lambda (g x) (g x)))
(the (nu (t₅₂) (-> t₅₂ t₅₂)) (lambda (x) x))
(the (nu (t₇₇) (-> t₇₇ t₇₇)) (lambda (x) x))
